### PR TITLE
[WIP] add relationship to customer -> customer group -> organization

### PR DESCRIPTION
**Pull request by flug**
@alexsegura just the beginning of the PR to see if it fits the criteria?

### CODE REVIEW

File: src/Resources/config/doctrine/Sylius.CustomerGroup.orm.xml [hunk 1]
```diff
@@ -0,0 +1,19 @@
+<?xml version="1.0" encoding="utf-8"?>
+<doctrine-mapping xmlns="http://doctrine-project.org/schemas/orm/doctrine-mapping"
+                  xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+                  xsi:schemaLocation="http://doctrine-project.org/schemas/orm/doctrine-mapping https://www.doctrine-project.org/schemas/orm/doctrine-mapping.xsd">
+  <entity name="AppBundle\Entity\Sylius\CustomerGroup" >
```
Comment (alexsegura) [inline src/Resources/config/doctrine/Sylius.CustomerGroup.orm.xml:5]: You need to add a `table` attribute for proper extension

```xml
<entity name="AppBundle\Entity\Sylius\CustomerGroup" table="sylius_customer_group">
```

File: app/DoctrineMigrations/Version20200730140157.php [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+    }
+
+    public function up(Schema $schema) : void
+    {
+        // this up() migration is auto-generated, please modify it to your needs
+        $this->abortIf($this->connection->getDatabasePlatform()->getName() !== 'postgresql', 'Migration can only be executed safely on \'postgresql\'.');
+
+        $this->addSql('CREATE TABLE organization (id SERIAL NOT NULL, group_id INT DEFAULT NULL, name TEXT NOT NULL, logo TEXT DEFAULT NULL, order_lead_time TIMESTAMP(0) WITHOUT TIME ZONE NOT NULL, limit_hour_order TEXT NOT NULL, start_hour_order TEXT NOT NULL, day_of_order_available TEXT NOT NULL, number_of_order_available TEXT NOT NULL, amount_of_subsidy_per_employee_and_order TEXT NOT NULL, coverage_of_delivery_costs_by_the_company_or_the_employee TEXT DEFAULT NULL, delivery_perimeter_expression TEXT NOT NULL, PRIMARY KEY(id))');
+        $this->addSql('CREATE UNIQUE INDEX UNIQ_C1EE637CFE54D947 ON organization (group_id)');
+        $this->addSql('CREATE TABLE customer_group (id SERIAL NOT NULL, code VARCHAR(255) NOT NULL, name VARCHAR(255) NOT NULL, PRIMARY KEY(id))');
```
Comment (alexsegura) [inline app/DoctrineMigrations/Version20200730140157.php:27]: Table `sylius_customer_group` is already there, there is no need to recreate it (see other comment about adding a `table` attribute to Doctrine mapping file)

File: src/Resources/config/doctrine/Sylius.Customer.orm.xml [hunk 3]
```diff
@@ -27,5 +27,15 @@
         </inverse-join-columns>
       </join-table>
     </many-to-many>
+    <many-to-many field="groups" target-entity="AppBundle\Entity\Sylius\CustomerGroup">
```
Comment (alexsegura) [inline src/Resources/config/doctrine/Sylius.Customer.orm.xml:30]: This mapping is [already managed by the superclass](https://github.com/Sylius/Sylius/blob/bbb6233c1d180fce360554254b902a164be4ce00/src/Sylius/Bundle/CustomerBundle/Resources/config/doctrine/model/Customer.orm.xml#L44-L46)

File: src/Resources/config/doctrine/Sylius.CustomerGroup.orm.xml [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+
+    <many-to-many field="customers" target-entity="AppBundle\Entity\Sylius\CustomerGroup">
```
Comment (alexsegura) [inline src/Resources/config/doctrine/Sylius.CustomerGroup.orm.xml:7]: Actually, I think you shouldn't extend CustomerGroup from Sylius. 
Looking at how it's done in Sylius, it's very simple: one customer can have many groups. Let's keep it like this and avoid extending too many things.

File: src/Resources/config/doctrine/Sylius.CustomerGroup.orm.xml [hunk 5]
```diff
@@ -0,0 +1,19 @@
...
+      <join-table name="customers_groups_organization">
```
Comment (alexsegura) [inline src/Resources/config/doctrine/Sylius.CustomerGroup.orm.xml:8]: The table name is not consistent with the relationship

File: src/Entity/Organization.php [hunk 6]
```diff
@@ -0,0 +1,132 @@
...
+    private $dayOfOrderAvailable;
+    private $numberOfOrderAvailable;
+    private $amountOfSubsidyPerEmployeeAndOrder;
+    private $coverageOfDeliveryCostsByTheCompanyOrTheEmployee;
+    /**
+     * @var string
+     *
+     * @Assert\Type(type="string")
+     */
+    protected $deliveryPerimeterExpression = 'distance < 3000';
```
Comment (alexsegura) [inline src/Entity/Organization.php:28]: I would prefer to have all these fields in an `OrganizationConfig` entity/table.

File: src/Entity/Organization.php [hunk 7]
```diff
@@ -0,0 +1,132 @@
...
+namespace AppBundle\Entity;
+
+use Symfony\Component\Validator\Constraints as Assert;
+
+class Organization
+{
+    private $id;
+    private $name;
+    private $address;
+    private $logo;
```
Comment (alexsegura) [inline src/Entity/Organization.php:13]: The `Organization` entity should be kept super simple (we would like to use it for other things in the future). 
Only those fields should be there.

File: src/Resources/config/doctrine/Sylius.CustomerGroup.orm.xml (hunk 5 above)
Comment (flug) [reply src/Resources/config/doctrine/Sylius.CustomerGroup.orm.xml:8]: okay but actually customer is a one-to-many and not many-to-many

File: src/Entity/Organization.php (hunk 6 above)
Comment (flug) [reply src/Entity/Organization.php:28]: ok

File: src/Entity/Organization.php (hunk 7 above)
Comment (flug) [reply src/Entity/Organization.php:13]: okay

File: src/Resources/config/doctrine/Sylius.Customer.orm.xml (hunk 3 above)
Comment (flug) [reply src/Resources/config/doctrine/Sylius.Customer.orm.xml:30]: it's not same relationship

**Pull request merged**